webpack/assets/javascripts/react_app/components/common/forms/CounterInput/CounterInput.js

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect, useState } from 'react';
+import RCInputNumber from 'rc-input-number';
+import PropTypes from 'prop-types';
+import { translate as __ } from '../../../../common/I18n';
+import { noop } from '../../../../common/helpers';
+
+const CounterInput = ({
+  value,
+  recommendedMaxValue,
+  maxValue,
+  onChange,
+  minValue,
+  disabled,
+  name,
+  id,
+  setError,
+  setWarning,
+}) => {
+  const [innerValue, setInnerValue] = useState(value);
+  useEffect(() => {
+    if (maxValue && innerValue > maxValue) {
+      setWarning(null);
+      setError(__('Specified value is higher than maximum value'));
+    } else if (recommendedMaxValue && innerValue > recommendedMaxValue) {
+      setError(null);
+      setWarning(__('Specified value is higher than recommended maximum'));
+    } else {
+      setError(null);
+      setWarning(null);
+    }
+  }, [recommendedMaxValue, maxValue, innerValue, setError, setWarning]);
+
+  const handleChange = v => {
+    setInnerValue(v);
+    onChange(v);
+  };
+
+  return (
+    <RCInputNumber
+      value={innerValue}
+      name={name}
+      id={id}
+      min={minValue}
+      disabled={disabled}
+      onChange={handleChange}
+      step={1}
```

Comment:
the `step` here is constant, though in some usages the step is given, i.e - https://github.com/theforeman/foreman_discovery/blob/19f3cdb8cd171749d98150e2d2f4a7304e93ec59/app/views/discovery_rules/_form.html.erb#L27

Reply:
Nice catch, I didn't realize that :+1: